### PR TITLE
Upgrade docker

### LINKED ISSUE #298: Multiple packages are missing in Docker image
In particular:

* `atari-py`
* `psutil`

Should `pip3 install` them all in `Dockerfile`.

Another report: https://www.coursera.org/learn/practical-rl/discussions/all/threads/vrpyQ9KzQJW6ckPSs-CVHw (says `imageio` and `skimage` are missing)

### PR DESCRIPTION
**Pull request by dmittov**
Fixes #298 and #317.

### CODE REVIEW

File: docker/run_jupyter.sh [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env bash
-jupyter notebook --no-browser --allow-root --ip 0.0.0.0
+jupyter notebook --no-browser --ip 0.0.0.0
```
Comment (dniku) [inline docker/run_jupyter.sh:3]: Could you remove this redundant empty line?

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,75 +1,61 @@
...
-                           torchvision \
-                           tensorflow \
-                           keras && \                           
-    python3 -m ipykernel.kernelspec
+USER jupyter
+SHELL ["/bin/bash", "-i", "-c"]
 
+RUN conda env create -f /tmp/deeplearning.yaml && \
+    conda activate deeplearning && \
+    python -m ipykernel install --user --name python3 --display-name "DeepLearning Python 3"
```
Comment (dniku) [inline docker/Dockerfile:52]: Could you also `rm /tmp/deeplearning.yaml`? I'm not sure if that's necessary, but just to be certain.

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,75 +1,61 @@
...
+FROM nvidia/cuda:10.2-base-ubuntu18.04 AS base-gpu
+
+FROM base-$device AS base
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get -qq update && \
+    apt-get install -y wget unzip git cmake xvfb sudo freeglut3-dev ffmpeg
+
+RUN adduser --disabled-password --gecos "Default user" jupyter && \
```
Comment (dniku) [inline docker/Dockerfile:14]: Not important by any measure, but would be nice to have this: could you make the username a build arg, like so

```
ARG USERNAME=jovyan
```

replacing `jupyter` with `jovyan`, like in [Jupyter Docker images](https://github.com/jupyter/docker-stacks/issues/358#issuecomment-288844834)?

As far as I see, after making that change, a simple `s/jupyter/$USERNAME/g` will fix the rest of the Dockerfile.

Comment (dmittov) [reply docker/Dockerfile]: fixed, check and resolve

File: docker/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmittov) [reply docker/Dockerfile]: fixed, check and resolve

Comment (dmittov) [reply docker/Dockerfile]: fixed, check and resolve

File: docker/run_jupyter.sh (hunk 1 above)
Comment (dmittov) [reply docker/run_jupyter.sh]: fixed, check and resolve

File: docker/Dockerfile [hunk 4]
```diff
@@ -1,75 +1,65 @@
...
+ARG device
+ARG username="jovyan"
+
+FROM ubuntu:18.04 AS base-cpu
+
+FROM nvidia/cuda:10.2-base-ubuntu18.04 AS base-gpu
+
+FROM base-$device AS base
+
+ARG username
```
Comment (dniku) [inline docker/Dockerfile:10]: Doesn't this override the default value specified before? (I'm not an expert on Dockerfiles.)

**Pull request merged**